# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Ud007it/Flyrank-ML-/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

My Lane Refresh / Content Opportunity Scoring lane and Why

I choose this lane because websites usually contain thousands of content pages, making it difficult for a content team to manually review every page. A ranking system can help identify which pages should be reviewed first based on search and content performance metrics. This makes the review process more efficient and helps the team focus on pages that have higher potential for improvement.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

Decision, Action, Cost

Research Question
How can search and content performance metrics be used to identify and rank webpages that should be prioritized for a content refresh?

Decision
Determining which specific content pages should be given priority for review each week.

Output
A ranked list of content pages ordered by their refresh priority score, along with human-readable reason codes.

Action
The SEO or content writing team can review the top-ranked pages in the queue and decide whether they need to be updated, expanded, pruned, or monitored.

Cost
A writer wastes 2-3 hours manually rewriting a page that was actually completely healthy, draining payroll budget.
A major traffic-driving page decays completely unnoticed, causing a permanent drop in organic search traffic and revenue.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [2]:
import pandas as pd


df = pd.read_csv("../../data/raw/content_refresh_anonymized.csv")

print("Dataset Shape:", df.shape)
print("-" * 30)
print("Average Impressions (90 Days):", round(df["impressions_90d"].mean(), 2))
print("Average Content Age (Days):", round(df["content_age_days"].mean(), 2))
print("Average Sessions (90 Days):", round(df["sessions_90d"].mean(), 2))
print("-" * 30)
print("Trend Direction Distribution:")
print(df["trend_direction"].value_counts())

Dataset Shape: (30000, 44)
------------------------------
Average Impressions (90 Days): 5200.37
Average Content Age (Days): 256.17
Average Sessions (90 Days): 37.07
------------------------------
Trend Direction Distribution:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64




Some key observations are:

 The dataset contains 30,000 rows and 44 columns.
 The average number of impressions over the last 90 days is approximately 5,200.37.
 The average content age is approximately 256.17 days.
 The average number of sessions over the last 90 days is approximately 37.07.

The dataset also contains pages with different performance trends. Out of 30,000 pages, 16,262 are marked as down, 5,962 as stable, 4,388 as up, 2,236 as new, and 1,152 as flat.

These observations suggest that the dataset contains a good variety of content performance patterns, making it suitable for building a system that helps prioritize pages for content review instead of relying on manual inspection.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*



This project is designed to support decision-making by identifying pages that may deserve further review based on observed search and content performance metrics.

The results should be considered as recommendations rather than final decisions. Human review is still required before making any content changes.

This project cannot prove that updating a page will improve its search ranking or increase traffic. It also cannot identify Google's ranking algorithm or establish a cause-and-effect relationship between content changes and performance.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.